unitbot.py

```diff
@@ -37,10 +39,32 @@ async def on_ready():
 
 @bot.event
 async def on_message(message): # Catches send messages and corrects non-SI units if neccesary. Most of the code behind this is in 'unitconversion.py'.
-    if bot.user.id is not message.author.id and message.author.bot is False and (message.guild is None or (message.guild is not None and discord.utils.get(message.guild.roles, name='imperial certified') not in message.author.roles)):
+    author = message.author
+    if bot.user.id is not author.id and author.bot is False:
+        user_roles = None
+        bot_roles = None
         processedMessage = unitconversion.process(message.content)
+        polite = False
+        if isinstance(message.channel, TextChannel):
+            for member in message.channel.members:
```

Comment:
You can actually get the roles of an message author directly without needing to iterate over all members. (possibly ten to hundreds of thousands in large servers, which might make a *lot* of API requests and noise)

Reply:
I thought that since the calls i used in potentially large loops were written in the API as properties rather than as async functions there wouldn't be excessive API calls; however, I could be wrong about that. I did see how you checked for imperial certified in the old version, and I liked it better.
However, I don't know how to use that same form of the API to get a `Member` object for the bot itself, and the wording of the API specifications wasn't completely clear to me that if a `Member` with the `id` of the `message.author` is contained in the `message.channel.members` list a `Member` object _must_ be returned by `message.author`, so while I was checking for the bot `Member` object, I figured I could check for the user as well.
I realize now that such a guarantee probably exists.
However, after a bit of searching through the documentation for the API I still couldn't find how to directly get the roles of the bot itself, given the message object.
Also, from my experience with the `discord.js` API, some degree of caching occurs between API requests? What degree, however, I do not know.
If we need a more efficient fix to this issue, I'll defer to someone who's better acquainted with the `discord.py` API and focus on manipulating strings instead.

Reply:
@Randall-Scharpf  try this one :)
```python
ctx = await bot.get_context(message)
bot_member = ctx.guild.get_member(bot.user.id)
roles = bot_member.roles
```

Reply:
I implemented it! Take a look, when you get a chance, at the new code in [this commit](https://github.com/Wendelstein7/DiscordUnitCorrector/pull/32/commits/34efa1e318b5211d051b67ba84c25b0561e30105).